### ISSUE TITLE
First abort the upload and warn them

### ISSUE THREAD

**Issue by hboetes** [nicotine-plus/nicotine-plus#1721]
@slook,

since you've already done some work on the leechers plugin.

Suppose the leech plugin first aborts the upload, explains why via a pmsg, and only after they reattempt bans the user?

Or would that be hard to implement?

**Comment by slook** [comment]
It wouldn't necessarily be that hard to implement, if the plugin users desire auto banning or other such functionality, then since the code is open source they are welcome to make modifications or write their own version of the leech detector plugin if they wish, and I have no problem with that.

Since I am not personally concerned with downloaders who wish to acquire a reasonable number of files using an account that "doesn't share enough", then I'm afraid that I have no motivation for implementing your suggested functionality.

My primary motivation for working on the plugin is to reduce the aggressiveness of it, and to reduce the inconvenience and confusion that is created towards new users on the network. Adding features to it which hinders transfers goes against this goal and seems like the opposite direction to where I would like the plugin to be (which is for it to be less annoying).

**Comment by hboetes** [comment]
ok.
